db/c.cc

```diff
@@ -2956,6 +2992,336 @@ void rocksdb_block_based_options_set_unpartitioned_pinning_tier(
+
+const char* rocksdb_externalfileingestioninfo_cf_name(
+    const rocksdb_externalfileingestioninfo_t* info, size_t* size) {
+  *size = info->rep.cf_name.size();
+  return info->rep.cf_name.data();
+}
+
+const char* rocksdb_externalfileingestioninfo_internal_file_path(
+    const rocksdb_externalfileingestioninfo_t* info, size_t* size) {
+  *size = info->rep.internal_file_path.size();
+  return info->rep.internal_file_path.data();
+}
+
+/* External write stall info */
+extern ROCKSDB_LIBRARY_API const char* rocksdb_writestallinfo_cf_name(
+    const rocksdb_writestallinfo_t* info, size_t* size) {
+  *size = info->rep.cf_name.size();
+  return info->rep.cf_name.data();
+}
+
+const rocksdb_writestallcondition_t* rocksdb_writestallinfo_cur(
+    const rocksdb_writestallinfo_t* info) {
+  return reinterpret_cast<const rocksdb_writestallcondition_t*>(
+      &info->rep.condition.cur);
+}
+
+const rocksdb_writestallcondition_t* rocksdb_writestallinfo_prev(
+    const rocksdb_writestallinfo_t* info) {
+  return reinterpret_cast<const rocksdb_writestallcondition_t*>(
+      &info->rep.condition.prev);
+}
+
+const char* rocksdb_memtableinfo_cf_name(const rocksdb_memtableinfo_t* info,
+                                         size_t* size) {
+  *size = info->rep.cf_name.size();
+  return info->rep.cf_name.data();
+}
+
+uint64_t rocksdb_memtableinfo_first_seqno(const rocksdb_memtableinfo_t* info) {
+  return info->rep.first_seqno;
+}
+uint64_t rocksdb_memtableinfo_earliest_seqno(
+    const rocksdb_memtableinfo_t* info) {
+  return info->rep.earliest_seqno;
+}
+uint64_t rocksdb_memtableinfo_num_entries(const rocksdb_memtableinfo_t* info) {
+  return info->rep.num_entries;
+}
+uint64_t rocksdb_memtableinfo_num_deletes(const rocksdb_memtableinfo_t* info) {
+  return info->rep.num_deletes;
+}
+
+/* event listener */
+
+struct rocksdb_eventlistener_t : public EventListener {
+  void* state_;
+  void (*destructor_)(void*);
+  void (*on_flush_begin)(void*, rocksdb_t*, const rocksdb_flushjobinfo_t*);
+  void (*on_flush_completed)(void*, rocksdb_t*, const rocksdb_flushjobinfo_t*);
+  void (*on_compaction_begin)(void*, rocksdb_t*,
+                              const rocksdb_compactionjobinfo_t*);
+  void (*on_compaction_completed)(void*, rocksdb_t*,
+                                  const rocksdb_compactionjobinfo_t*);
+  void (*on_subcompaction_begin)(void*, const rocksdb_subcompactionjobinfo_t*);
+  void (*on_subcompaction_completed)(void*,
+                                     const rocksdb_subcompactionjobinfo_t*);
+  void (*on_external_file_ingested)(void*, rocksdb_t*,
+                                    const rocksdb_externalfileingestioninfo_t*);
+  void (*on_background_error)(void*, uint32_t, rocksdb_status_ptr_t*);
+  void (*on_stall_conditions_changed)(void*, const rocksdb_writestallinfo_t*);
+  void (*on_memtable_sealed)(void*, const rocksdb_memtableinfo_t*);
+
+  virtual void OnFlushBegin(DB* db, const FlushJobInfo& info) {
```

Comment:
Overrides should be marked explicitly.
OnFlushBegin overrides methods from EventListener but is not annotated with override.

same for methods below.